### PR TITLE
Auto unlock wallet app

### PR DESCRIPTION
**Pull request by arjunu**

### CODE REVIEW

File: packages/wallet/app/App.tsx [hunk 1]
```diff
@@ -116,7 +124,10 @@ const AppLayout: React.FunctionComponent<TLayoutProps> = ({
       });
     }
 
-    if (previousInitStatus === EInitStatus.OFFLINE && initStatus === EInitStatus.IN_PROGRESS) {
+    if (
+      [EInitStatus.NOT_STARTED, EInitStatus.OFFLINE].includes(previousInitStatus) &&
+      initStatus === EInitStatus.IN_PROGRESS
```
Comment (arjunu) [inline packages/wallet/app/App.tsx:129]: there is still sometimes a white screen because of the delay between app opening and this code executing.

Comment (arjunu) [reply packages/wallet/app/App.tsx:130]: ok I updated `react-native-bootsplash` and the white screen seems to be gone now

File: packages/wallet/app/App.tsx [hunk 2]
```diff
@@ -104,21 +106,33 @@ const AppLayout: React.FunctionComponent<TLayoutProps> = ({
...
+    };
+  }, [start, reset]);
 
   React.useEffect(() => {
     if (previousInitStatus === EInitStatus.IN_PROGRESS) {
       // wait for all interactions to finish before showing app
       // otherwise for a moment a white screen will appear for a user
       // eslint-disable-next-line @typescript-eslint/no-floating-promises
       InteractionManager.runAfterInteractions(() => {
+        // eslint-disable-next-line @typescript-eslint/no-floating-promises
```
Comment (arjunu) [inline packages/wallet/app/App.tsx:123]: because now `show` and `hide` return promises

File: packages/wallet/app/App.tsx [hunk 3]
```diff
@@ -104,21 +106,33 @@ const AppLayout: React.FunctionComponent<TLayoutProps> = ({
 
   React.useEffect(() => {
     start();
-  }, [start]);
+
+    return () => {
+      // on Android when app is reopened after back button minimize, App remounts and starts init
+      // so on unmount reset init status flag for sagas
+      reset();
```
Comment (desfero) [inline packages/wallet/app/App.tsx:113]: cool

File: packages/wallet/app/modules/auth/sagas.ts [hunk 4]
```diff
@@ -1,26 +1,25 @@
...
-  assertNever,
+  toEthereumPrivateKey,
 } from "@neufund/shared-utils";
 // eslint-disable-next-line react-native/split-platform-components
 import { ToastAndroid } from "react-native";
-import Config from "react-native-config";
 
 import { biometricsModuleApi } from "modules/biometrics/module";
-import { walletEthModuleApi, EWalletExistenceStatus } from "modules/eth/module";
+import { EWalletExistenceStatus, walletEthModuleApi } from "modules/eth/module";
```
Comment (desfero) [inline packages/wallet/app/modules/auth/sagas.ts:14]: hm, i'm curious why it sorted that now and not before?

File: packages/wallet/app/modules/auth/sagas.ts [hunk 5]
```diff
@@ -42,23 +41,11 @@ export function* trySignInExistingAccount(): SagaGenerator<void> {
...
-          break;
-
-        case "auto":
-        default:
-          yield* call(signInExistingAccount);
-          break;
-      }
-
+      yield* call(allowToUnlockExistingAccount);
+      yield put(authActions.unlockAccount());
```
Comment (desfero) [inline packages/wallet/app/modules/auth/sagas.ts:45]: let's add a comment why we have both `yield* call(allowToUnlockExistingAccount);` and `yield put(authActions.unlockAccount());` as i was not sure initially why that was needed

File: packages/wallet/storybook/index.tsx [hunk 6]
```diff
@@ -30,6 +30,7 @@ const Storybook: React.FunctionComponent = () => {
     InteractionManager.runAfterInteractions(() => {
       // we do have custom logic when to hide splash screen for the normal flow
       // for storybook we can hide just when we have UI ready
+      // eslint-disable-next-line @typescript-eslint/no-floating-promises
       RNBootSplash.hide();
```
Comment (desfero) [inline packages/wallet/storybook/index.tsx:34]: let's also use `async/await` here

**Review by desfero — CHANGES_REQUESTED** [review]
Some small comments

File: packages/wallet/storybook/index.tsx (hunk 6 above)
Comment (arjunu) [reply packages/wallet/storybook/index.tsx]: done

File: packages/wallet/app/modules/auth/sagas.ts (hunk 4 above)
Comment (arjunu) [reply packages/wallet/app/modules/auth/sagas.ts:14]: strange indeed

File: packages/wallet/app/App.tsx (hunk 2 above)
Comment (arjunu) [reply packages/wallet/app/App.tsx]: done

File: packages/wallet/app/modules/auth/sagas.ts (hunk 5 above)
Comment (arjunu) [reply packages/wallet/app/modules/auth/sagas.ts:45]: done

**Review by desfero — APPROVED** [review]

**Pull request merged**